components/status_dropdown/status_dropdown.tsx

```diff
@@ -24,76 +23,81 @@ import StatusOfflineIcon from 'components/widgets/icons/status_offline_icon';
 import OverlayTrigger from 'components/overlay_trigger';
 import CustomStatusText from 'components/custom_status/custom_status_text';
 
+import {ActionFunc} from 'mattermost-redux/types/actions';
+
+import {UserCustomStatus, UserStatus} from 'mattermost-redux/types/users';
+
 import './status_dropdown.scss';
 
-export default class StatusDropdown extends React.PureComponent {
-    static propTypes = {
-        style: PropTypes.object,
-        status: PropTypes.string,
-        userId: PropTypes.string.isRequired,
-        profilePicture: PropTypes.string,
-        autoResetPref: PropTypes.string,
-        actions: PropTypes.shape({
-            openModal: PropTypes.func.isRequired,
-            setStatus: PropTypes.func.isRequired,
-            unsetCustomStatus: PropTypes.func.isRequired,
-            setStatusDropdown: PropTypes.func.isRequired,
-        }).isRequired,
-        customStatus: PropTypes.object,
-        isCustomStatusEnabled: PropTypes.bool.isRequired,
-        isStatusDropdownOpen: PropTypes.bool.isRequired,
-        showCustomStatusPulsatingDot: PropTypes.bool.isRequired,
-    }
+type Props = {
+    status?: string;
+    userId: string;
+    profilePicture: string;
+    autoResetPref?: string;
+    actions: {
+        openModal: (modalData: {modalId: string; dialogType: any; dialogProps?: any}) => void;
+        setStatus: (status: UserStatus) => ActionFunc;
+        unsetCustomStatus: () => ActionFunc;
+        setStatusDropdown: (open: boolean) => void;
+    };
+    customStatus: UserCustomStatus | Record<string, any>;
+    isCustomStatusEnabled: boolean;
+    isStatusDropdownOpen: boolean;
+    showCustomStatusPulsatingDot: boolean;
+}
 
+export default class StatusDropdown extends React.PureComponent <Props> {
     static defaultProps = {
         userId: '',
         profilePicture: '',
         status: UserStatuses.OFFLINE,
         customStatus: {},
     }
 
-    isUserOutOfOffice = () => {
+    isUserOutOfOffice = (): boolean => {
         return this.props.status === UserStatuses.OUT_OF_OFFICE;
     }
 
-    setStatus = (status) => {
+    setStatus = (status: string): void => {
         this.props.actions.setStatus({
             user_id: this.props.userId,
             status,
+            manual: false,
```

Comment:
hi @lieut-data is this code line already ok or better I set UserStatus to be like this ?
```
// from mattermost-redux/types/users
export type UserStatus = {
    user_id: string;
    status: string;
    manual?: boolean;
    last_activity_at?: number;
    active_channel?: string;
};

```

Reply:
I /think/ this is ok, but will defer to @devinbinnie.

Reply:
I think it's better to fix the type if the function doesn't expect all of the parameters to be there.

Reply:
Hi @devinbinnie I already alter the type UserStatus.